app/test/test_ftp.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+#   encoding: UTF-8
+
+import os
+import multiprocessing
+import random
+import shutil
+import sys
+import tempfile
+import time
+import unittest
+
+# To run test in CF
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
+print(sys.path)
+
+from ftpclient import FTPWorker
+import localserver
+
+"""
+To run this test in a Cloudfoundry environment:
+
+$ cf push seft-publisher-test
+$ cf logs seft-publisher-test --recent
+
+"""
+
+
+class NeedsTemporaryDirectory():
+
+    def setUp(self):
+        self.root = tempfile.mkdtemp()
```

Comment:
It's a test for an FTP server, so it needs a place to keep its files. I think that's legitimate in this case. If CF can't support that, its a deficiency in CF and someone should fix that.